src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.Icu.cs

```diff
@@ -292,6 +301,15 @@ private unsafe int IndexOfOrdinalHelper(ReadOnlySpan<char> source, ReadOnlySpan<
                         throw new Exception((string)ex_result);
                     return result;
                 }
+#elif TARGET_OSX || TARGET_MACCATALYST || TARGET_IOS || TARGET_TVOS
+                if (GlobalizationMode.Hybrid)
+                {
+                    NSRange result = Interop.Globalization.IndexOfNative(m_name, m_name.Length, b, target.Length, a, source.Length, options, fromBeginning);
+                    if (matchLengthPtr == null)
+                       matchLengthPtr = &result.Length;
+                    *matchLengthPtr = result.Length;
```

Comment:
I'm not sure if we should be reassigning `matchLengthPtr` if its `null`.
We can probably match
```
if (matchLengthPtr != null)
    *matchLengthPtr = result.Length;
```

Reply:
If we passed null as `matchLengthPtr`, e.g. https://github.com/dotnet/runtime/blob/a7ab5725e01bd348d3cb6ea8fc25fc96fdbe3e61/src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.cs#L988
then I believe we don't expect it to get re-assigned and it would be surprising if the function overwrites the null-pointer.

Reply:
Yes you are right, updated.